src/solvers/changeCobraSolver.m

```diff
@@ -1,365 +1,385 @@
+%                     but best avoided for all solvers
+%     qpng            qpng QP solver with Matlab mex interface (in glpkmex
+%                     package, only limited support for small problems)
+%
+%     legacy solvers:
+%
+%     gurobi5         Gurobi 5.0 accessed through built-in Matlab mex interface
+%     gurobi_mex      Gurobi accessed through Matlab mex interface (Gurobi mex)
+%
+% Currently allowed MIQP solvers:
+%
+%     fully supported solvers:
+%
+%     cplex_direct    CPLEX accessed directly through Tomlab cplex.m. This gives
+%                     the user more control of solver parameters. e.g.
+%                     minimising the Euclidean norm of the internal flux to
+%                     get rid of net flux around loops
+%     gurobi6         Gurobi 6.* accessed through built-in Matlab mex interface
+%     gurobi7         Gurobi 7.* accessed through built-in Matlab mex interface
+%     ibm_cplex       The IBM API for CPLEX using the CPLEX class
+%     tomlab_cplex    CPLEX MIQP solver accessed through Tomlab environment
+%
+%     legacy solvers:
+%
+%     gurobi5         Gurobi 5.0 accessed through built-in Matlab mex interface
+%     gurobi_mex      Gurobi accessed through Matlab mex interface (Gurobi mex)
+%
+% Currently allowed NLP solvers:
+%
+%     fully supported solvers:
+%
+%     matlab          MATLAB's fmincon.m
+%
+%     experimental support:
+%
+%     tomlab_snopt    SNOPT solver accessed through Tomlab environment
+%
+% It is a good idea to put this function call into your startup.m file
+% (usually matlabinstall/toolboxes/local/startup.m)
+%
+% Original file: Markus Herrgard 1/19/07
+
+global SOLVERS;
+global OPT_PROB_TYPES;
+global CBT_LP_SOLVER;
+global CBT_MILP_SOLVER;
+global CBT_QP_SOLVER;
+global CBT_MIQP_SOLVER;
+global CBT_NLP_SOLVER;
+global ENV_VARS;
+global TOMLAB_PATH;
+
+if isempty(SOLVERS) || isempty(OPT_PROB_TYPES)
+    ENV_VARS.printLevel = false;
+    initCobraToolbox;
+end
+
+% configure the environment variables
+configEnvVars();
+
+% Print out all solvers defined in global variables CBT_*_SOLVER
+if nargin < 1
+    definedSolvers = [CBT_LP_SOLVER, CBT_MILP_SOLVER, CBT_QP_SOLVER, CBT_MIQP_SOLVER, CBT_NLP_SOLVER];
+    if isempty(definedSolvers)
+        fprintf('No solvers are defined!\n');
+    else
+        fprintf('Defined solvers are:\n');
+        for i = 1:length(OPT_PROB_TYPES)
+            varName = horzcat(['CBT_', OPT_PROB_TYPES{i}, '_SOLVER']);
+            if ~isempty(eval(varName))
+                fprintf('    %s: %s\n', varName, eval(varName));
+            end
+        end
+    end
+    return;
+end
+
+if nargin < 2
+    solverType = 'LP';
+else
+    solverType = upper(solverType);
+end
+
+if nargin < 3
+    printLevel = 1;
+end
+
+% Attempt to set the user provided solver for all optimization problem types
+if (strcmp(solverType, 'ALL'))
+    for i = 1:length(OPT_PROB_TYPES)
+        changeCobraSolver(solverName, OPT_PROB_TYPES{i}, 0);
```

Comment:
But doesn't this throw an error, if you run it with all and an output value?

And wouldn't it be sufficient to call it with a 0 printlevel in the init Script to avoid this?